.github/workflows/docker.yml

```diff
@@ -57,3 +57,50 @@ jobs:
           push: true
           tags: geodynamics/aspect:${{github.ref_name}}
 
+  build-docker-tacc:
+    runs-on: ubuntu-latest
+    if: github.repository == 'geodynamics/aspect'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_LOGIN }}
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push Docker image for main
+        if: contains(github.event_name, 'push')
```

Comment:
What does this do? only react to pushes? what about merges?

Reply:
shouldn't this be only on the main branch?

Reply:
The 'push' event triggers whenever commits are pushed to a branch. A merge counts as a push. There is no separate merge event (not clearly specified, this is all the docs say: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push). We ensure only pushes to 'main' are considered by specifying a branch in the 'on:push' section at the top of the workflow.  This logic is the same as the one further up in the file that creates our regular (non-TACC) docker images.
The main reason I need this 'if' here is to separate pushes to main from releases to give different tags to the created images.

I am not 100% sure this workflow will work right away, because I dont know if the login to ghcr.io will work. However, I think the best way to test it is to try it. The worst things that can happen is that we have one workflow that fails (it will not show up on PRs, because it is not triggered by PRs).

Reply:
sure.